### PR TITLE
Edit/Remove formQuestion from formTemplate backend

### PR DESCRIPTION
**Pull request by dhmi19**
## Notion ticket link

[Form Management: Edit/remove question in form template](https://www.notion.so/uwblueprintexecs/Task-Board-F22-9e03bf9f7c4343f09a8a5c16dc48a12a?p=792e8a8c96954713a7399446547c0fad&pm=c)

## Implementation description
* made a new endpoint for editing a question in the form template
* made a new function in the admin service to "edit" the form question

## Steps to test
1. Make a PATCH request to `/admin/formTemplate/formQuestion/:oldQuestionId` <-- get this form mongoDB 
2. use postman with the following body:
```
{
    "newFormQuestion" : {
        "category": "CampSpecific",
        "type" : "MultipleChoice",
        "question": "Can the camper swim?",
        "required" : false,
        "description" : "We will not have life guards on duty",
        "options": ["Yes", "No"]
    }
}
```
3. You should get `true` if the oldQuestionId you provided was in the template

4. Make sure to test error paths (no auth, invalid body, params)

## What should reviewers focus on?
* testing
* use of transactions?

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/middlewares/validators/adminValidators.ts [hunk 1]
```diff
@@ -60,3 +60,39 @@ export const formTemplateAddQuestionValidator = async (
...
+  if (
+    !req.params.oldQuestionId ||
+    !validatePrimitive(req.params.oldQuestionId, "string")
+  ) {
+    return res.status(400).send("oldQuestionId param must be a string");
+  }
+
+  if (
+    !req.body.newFormQuestion ||
+    !validateFormQuestion(req.body.newFormQuestion)
```
Comment (e-wai) [inline backend/typescript/middlewares/validators/adminValidators.ts:92]: For editing questions, do users have to edit every field at once? What if they want to edit one field?

File: backend/typescript/services/implementations/adminService.ts [hunk 2]
```diff
@@ -163,6 +163,73 @@ class AdminService implements IAdminService {
...
+      // Get the formTemplate which has the oldQuestionId in its list of questions
+      const formTemplate = await MgFormTemplate.findOne(
+        { formQuestions: oldQuestionId },
+        {},
+        { session },
+      );
+
+      if (!formTemplate) {
+        throw new Error(
+          "Failed to retrieve the form template with that question",
```
Comment (e-wai) [inline backend/typescript/services/implementations/adminService.ts:205]:
```suggestion
          `Failed to retrieve the form template, no template with question of id: ${oldQuestionId}`,
```

**Review by e-wai — CHANGES_REQUESTED** [review]
Nothing major, functionality looks good to me

File: backend/typescript/middlewares/validators/adminValidators.ts (hunk 1 above)
Comment (dhmi19) [reply backend/typescript/middlewares/validators/adminValidators.ts:92]: The rest of the fields would remain the same as the previous data. It's not super restful but we are technically not PATCHing so I thought it was fine?

Comment (e-wai) [reply backend/typescript/middlewares/validators/adminValidators.ts:92]: Yeah I mean I'm not especially strongly opinionated here; I can see why in our use case it's easiest to just send the entire question back

**Review by e-wai — APPROVED** [review]
🥂

**Pull request merged**